topdown/json_bench_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package topdown
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/storage"
+	inmem "github.com/open-policy-agent/opa/storage/inmem/test"
+)
+
+func BenchmarkJSONFilterShallow(b *testing.B) {
+	ctx := context.Background()
+
+	sizes := []int{10, 100, 1000, 10000}
+
+	for _, n := range sizes {
+		source := genTestObject(n)
+		for _, m := range sizes {
+			if m > n {
+				continue // skip tests where too many paths would be removed. (error case)
+			}
+			testName := fmt.Sprintf("object-%d-%d", n, m)
+			// Build dataset right before use:
+			paths := make([]*ast.Term, 0, m)
+			// paths to remove
+			for i := 0; i < m; i++ {
+				paths = append(paths, ast.StringTerm("/"+strconv.FormatInt(int64(i), 10)))
+			}
+			pathList := ast.NewArray(paths...)
+
+			b.ResetTimer()
+			b.Run(testName, func(b *testing.B) {
+				runJSONFilterBenchmarkTest(ctx, b, source, pathList)
+			})
+		}
+	}
+}
+
+func BenchmarkJSONFilterNested(b *testing.B) {
+	ctx := context.Background()
+
+	sizes := []int{10, 100, 1000, 10000}
+
+	for _, n := range sizes {
+		source := genNestedTestObject(n, 3)
+		for _, m := range sizes {
+			if m > n {
+				continue // skip tests where too many paths would be removed. (error case)
+			}
+			testName := fmt.Sprintf("object-%d-%d", n, m)
+			// Build dataset right before use:
+			paths := make([]*ast.Term, 0, m)
+			// paths to remove
+			for i := 0; i < m; i++ {
+				idx := strconv.FormatInt(int64(i), 10)
+				paths = append(paths, ast.StringTerm("/"+idx+"/"+idx+"/"+idx))
+			}
+			pathList := ast.NewArray(paths...)
+
+			b.ResetTimer()
```

Comment:
[nit] Isn't doing this inside `runJSONFilterBenchmarkTest` sufficient?

Reply:
Ah, that's definitely a copy/paste holdover from an earlier iteration. I'll patch that shortly!